### PR TITLE
TF-TRT test SquaredDiff op conversion in dynamic shape mode

### PR DESCRIPTION
**Pull request by tfeher**
This PR adds  explicit batch and dynamic shape tests to the SquaredDiff op converter. Tagging @bixia1 for review.

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -6466,87 +6466,68 @@ NodeDef GetSquaredDifferenceNodeDef(DataType dtype) {
...
+      {/*dims_x=*/{1, 2, 3},
+       /*dims_y=*/{1, 7, 5},
+       /*value_x=*/common_input,
+       /*value_y=*/std::vector<float>(7 * 5, 0),
+       /*expected_output_dims=*/{1, 1, 2, 3},
+       /*expected_output=*/common_input,
+       trt_mode == TrtTestMode::kDynamicShape
+           ? Status::OK()
+           : Status(error::INVALID_ARGUMENT, "Infeasible broadcast scheme"),
+       Status(error::INVALID_ARGUMENT, "Negative number of dimensions -1")},
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:6503]: Do you know why we delay the handling of error until [here](https://github.com/tensorflow/tensorflow/blob/a26381f3dc49cfe7ef4bdc05652fc71b62f932f1/tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc#L43) returns dims.nbDims == -1? 
I see that there is an error in building the engine before that:
 trt_logger.cc:42] DefaultLogger (Unnamed Layer* 5) [Shuffle]: reshaping failed for tensor: input

**Review by tfeher — COMMENTED** [review]
Thanks @bixia1 for the review, I have answered the question.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 1 above)
Comment (bixia1) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:6479]: Maybe we should not pass nbDims==-1 to [here](https://github.com/tensorflow/tensorflow/blob/a26381f3dc49cfe7ef4bdc05652fc71b62f932f1/tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc#L54).  Do you think we should return an error before calling TrtDimsToTensorShape, that says either binding_index is out of bound or the input shapes do not lead to a valid TRT engine, and put a comment  explaining TRT current doesn't report an error in engine build time?

Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: That is a good idea, I have implemented it.

**Review by bixia1 — COMMENTED** [review]
Can you squash the commits into one? I will take care of the merge after.

**Review by bixia1 — APPROVED** [review]

**Pull request merged**